src/Analyzers/CSharp/CodeFixes/UseThrowExpression/UseThrowExpressionCodeFixProvider.cs

```diff
@@ -52,14 +55,37 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                 var ifStatement = root.FindNode(diagnostic.AdditionalLocations[0].SourceSpan);
                 var throwStatementExpression = root.FindNode(diagnostic.AdditionalLocations[1].SourceSpan);
                 var assignmentValue = root.FindNode(diagnostic.AdditionalLocations[2].SourceSpan);
+                var expressionStatement = assignmentValue.GetAncestor<ExpressionStatementSyntax>();
+                var identifierName = expressionStatement.DescendantNodes().ElementAt(1);
 
-                // First, remote the if-statement entirely.
+                var block = ((IfStatementSyntax)ifStatement).Statement.ChildNodes().FirstOrDefault();
+                var closeBrace = ((IfStatementSyntax)ifStatement).CloseParenToken;
+
+                var triviaList = new SyntaxTriviaList()
+                    .AddRange(closeBrace.GetAllTrailingTrivia().Where(t => !t.IsWhitespaceOrEndOfLine()))
+                    .AddRange(block.GetLeadingTrivia().Where(t => !t.IsWhitespaceOrEndOfLine()))
+                    .AddRange(block.GetTrailingTrivia().Where(t => !t.IsWhitespaceOrEndOfLine()));
+
+                // First, remove the if-statement entirely.
                 editor.RemoveNode(ifStatement);
 
                 // Now, update the assignment value to go from 'a' to 'a ?? throw ...'.
-                editor.ReplaceNode(assignmentValue,
-                    generator.CoalesceExpression(assignmentValue,
-                    generator.ThrowExpression(throwStatementExpression)));
+                if (triviaList.IsEmpty())
+                {
+                    editor.ReplaceNode(assignmentValue,
+                        generator.CoalesceExpression(assignmentValue,
+                        generator.ThrowExpression(throwStatementExpression)));
+                }
+                else
+                {
+                    triviaList = triviaList.AddRange(block.GetTrailingTrivia().Where(t => t.IsEndOfLine()));
+
+                    var newExpression = generator.ExpressionStatement(generator.AssignmentStatement(identifierName,
```

Comment:
I think we should try not to generate the new code differently even when we have to update trivia.  

Can you remind me why we can't add the trivia onto the node returned by `generator.CoalesceExpression`?  Was it because the assignment doesn't contain the semicolon so the trivia gets added before the semicolon token?  If that is the case we likely just need to find the right node to add it onto in the new tree.

Reply:
Yes, if you add the trivia to the coalesce expression it gets added before the semicolon token